### PR TITLE
Missing call to slave.terminate()

### PR DESCRIPTION
**Pull request by carlossg**
Apparently it is needed or jenkins will think failed nodes are still available in some cases,
not calling the plugin again to provision nodes,
thus leaving jobs stuck waiting for executor that will never be created

Based on yet-another-docker-plugin which was mentioned by @stephenc as a good example of cloud api implementation https://github.com/KostyaSha/yet-another-docker-plugin/blob/db0026ea46b1ca24a03c4212d09acefff0b1f2e4/yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/launcher/DockerComputerJNLPLauncher.java#L281

### CODE REVIEW

**Comment by carlossg** [comment]
@reviewbybees

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java [hunk 1]
```diff
@@ -222,15 +222,16 @@ public void launch(SlaveComputer computer, TaskListener listener) {
                 if (containerStatuses != null) {
                     logLastLines(containerStatuses, podId, namespace, slave, null, client);
                 }
+                slave.terminate();
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java]: Is this necessary? it's already called in the catch block

**Comment by burnMyDread** [comment]
Do we want to test this in an installed cluster before merging it?

**Comment by carlossg** [comment]
The tests are run in my GKE cluster https://jenkins.gke.csanchez.org/blue/organizations/jenkins/jenkinsci%2Fkubernetes-plugin/detail/node-removal/1/

I'll go through some more testing before merging

**Comment by stephenc** [comment]
Technically oc-cloud is the best implementation... and Carlos should be able to see that, but there are some specifics to CJOC that may mean it is not as applicable to this plugin

**Pull request merged**